### PR TITLE
style: add comments

### PR DESCRIPTION
**Pull request by mchlkucera**
Pls @immakermatty fact-check.

To @edgar0011 @jankrakovsky do these comments make sense?

### CODE REVIEW

**Review by edgar0011 — APPROVED** [review]

File: Spectoda.js [hunk 1]
```diff
@@ -1269,15 +1313,8 @@ export class Spectoda {
...
-  // event_label example: "evt1"
-  // event_value example: 100.0
   /**
-   *
-   * @param {*} event_label
-   * @param {*} event_value
-   * @param {number|number[]} device_ids
-   * @param {*} force_delivery
-   * @returns
+   * E.g. event "brigh" to value 1234
```
Comment (immakermatty) [inline Spectoda.js:1317]: range for percentage is from -100.0 to 100.0  - not value 1234

File: Spectoda.js [hunk 2]
```diff
@@ -1381,6 +1414,9 @@ export class Spectoda {
     return this.syncTimeline();
   }
 
+  /**
+   * TODO MATTY
```
Comment (immakermatty) [inline Spectoda.js:1418]: syncTimeline synchronizes the timeline to Controller from App

File: Spectoda.js [hunk 3]
```diff
@@ -1399,16 +1435,9 @@ export class Spectoda {
...
-
-  //   this.#resetClockSyncInterval();
-
-  //   return this.runtime.syncClock().then(() => {
-  //     logging.debug("> App clock synchronized");
-  //   });
-  // }
-
+  /**
+   * TODO MATTY
```
Comment (immakermatty) [inline Spectoda.js:1439]: syncState() synchronizes TNGL variable state of given ID to all other IDs

File: Spectoda.js [hunk 4]
```diff
@@ -1589,6 +1628,10 @@ export class Spectoda {
       });
   }
 
+  /**
+   * For FW nerds
+   * TODO MATTY
```
Comment (immakermatty) [inline Spectoda.js:1633]: tells the connected Controller to update a peer Controller with its own firmware.

File: Spectoda.js [hunk 5]
```diff
@@ -1985,6 +2052,10 @@ export class Spectoda {
     return this.runtime.execute(payload, null);
   }
 
+  /**
+   * @deprecated
+   * TODO MATTY is this true?
```
Comment (immakermatty) [inline Spectoda.js:2057]: Deprecated, please remove

File: Spectoda.js [hunk 6]
```diff
@@ -2023,6 +2094,10 @@ export class Spectoda {
     });
   }
 
+  /**
+   * @deprecated
+   * TODO MATTY is this true?
```
Comment (immakermatty) [inline Spectoda.js:2099]: Deprecated. Will be replaced by IO operations

File: Spectoda.js [hunk 7]
```diff
@@ -2127,6 +2207,10 @@ export class Spectoda {
     });
   }
 
+  /**
+   * Tells the currently `connect`ed controller to synchronize its event state history with other controllers in the network.
```
Comment (immakermatty) [inline Spectoda.js:2211]: Nope.

Synchronizes the EventHistory from connected Controller into App.

File: Spectoda.js [hunk 8]
```diff
@@ -2171,6 +2255,11 @@ export class Spectoda {
     });
   }
 
+  /**
+   * ! Useful
+   * Erases the event state history of ALL CONTROLLERS in the network Spectoda.js is `connect`ed to.
+   * TODO This should be called `eraseEventStateHistory`
```
Comment (immakermatty) [inline Spectoda.js:2261]: eraseEventStates

File: Spectoda.js [hunk 9]
```diff
@@ -2196,6 +2293,9 @@ export class Spectoda {
     return this.runtime.execute(payload, null);
   }
 
+  /**
+   * For FW Nerds
```
Comment (immakermatty) [inline Spectoda.js:2297]: forces a TNGL variable state save on the connected controller. TNGL variable state is by default saved every 8 seconds after no event is emitted

File: Spectoda.js [hunk 10]
```diff
@@ -2204,6 +2304,10 @@ export class Spectoda {
     return this.runtime.execute(payload, null);
   }
 
+  /**
+   * TODO MATTY
+   * Seems a duplicate of `readControllerCodes`
```
Comment (immakermatty) [inline Spectoda.js:2309]: Refer to 0.13-dev branch were I refactored all names of Spectoda.js functions

File: Spectoda.js [hunk 11]
```diff
@@ -2382,6 +2499,9 @@ export class Spectoda {
     });
   }
 
+  /**
+   * For FW nerds
```
Comment (immakermatty) [inline Spectoda.js:2503]: Reads the TNGL variable on given ID from App WASM

File: Spectoda.js [hunk 12]
```diff
@@ -2563,17 +2708,25 @@ export class Spectoda {
...
+   */
   execute(bytecode) {
     return this.runtime.execute(bytecode, null, 60000);
   }
 
   // emits JS event
+  /**
+   * TODO MATTY
```
Comment (immakermatty) [inline Spectoda.js:2720]: Private function. Emit JS events like "connected" or "eventstateupdates".

File: Spectoda.js [hunk 13]
```diff
@@ -2625,6 +2783,13 @@ export class Spectoda {
     return this.runtime.execute(command_bytes, null);
   }
 
+  /**
+   * TNGL BANKS: A concept in which you can save Tngl to different memory banks, and then load them when you need. Used to speed up tngl synchronization in installations where all animations don't fit to one Tngl file
+   */
+
+  /**
+   * Save the current uploaded Tngl (via `uploadTngl) to the bank in parameter
```
Comment (immakermatty) [inline Spectoda.js:2791]: uploadTngl() is now called writeTngl() but should be called loadTngl()

File: Spectoda.js [hunk 14]
```diff
@@ -1119,6 +1185,11 @@ export class Spectoda {
     });
   }
 
+  /**
+   * ! Useful
+   * Writes the given TNGL code to the controller.
```
Comment (immakermatty) [inline Spectoda.js:1190]: Loads the given TNGL code to the controller.

should be called loadTngl()

File: Spectoda.js [hunk 15]
```diff
@@ -520,56 +586,35 @@ export class Spectoda {
    * all events: event.target === the sender object (SpectodaWebBluetoothConnector)
    * event "disconnected": event.reason has a string with a disconnect reason
    *
+   * TODO I think this should expose an "off" method to remove the listener
```
[the remark this answers is not in the document]
Comment (mchlkucera) [reply Spectoda.js:589]: @edgar0011 Is it ok not having an `off` function?

File: Spectoda.js (hunk 3 above)
Comment (mchlkucera) [reply Spectoda.js:1439]: What's a TNGL variable state

File: Spectoda.js [hunk 16]
```diff
@@ -1878,6 +1929,10 @@ export class Spectoda {
     });
   }
 
+  /**
+   * ! Useful
+   * Puts ALL CONTROLLERS in the network into the DEFAULT network. More info at the top of this file.
```
[the remark this answers is not in the document]
Comment (mchlkucera) [reply Spectoda.js:1934]: Ok, so can we just get rid of the name "default network" altogether? Lets call it "controller is not in a network" to simplify this?

File: Spectoda.js (hunk 12 above)
Comment (mchlkucera) [reply Spectoda.js:2720]: Why is it not marked as private?

**Pull request merged**